### ISSUE TITLE
Automatically terminate Experiments without durations in Rollout steps

### ISSUE THREAD

**Issue by kcboschert** [argoproj/argo-rollouts#900]
# Summary

Experiments attached to Rollout steps currently either:
1. run for the specified `duration`
2. run until a user manually terminates the experiment

We'd like to avoid defining an arbitrary duration for our use case below and have the Experiment terminate automatically once the analyses attached to the experiment have completed.

# Use Cases

We're starting an experiment during our canary rollout to run a smoke test suite against a canary pod before having it take live traffic:
```yaml
strategy:
  canary:
    steps:
      - experiment:
          templates:
          - name: smoke-target
            specRef: canary
          analyses:
          - templateName: smoke-analysis
            name: smoke-suite
```
We'd like the experiment to automatically terminate once the `smoke-suite` analysis completes.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by jessesuen** [comment]
Duplicate of https://github.com/argoproj/argo-rollouts/issues/878

This actually a bug. We have a field which is supposed to allow for this, but it doesn't appear to be working

**Comment by jessesuen** [comment]
Fixed in v0.10.2 (now released)
